src/controller.js

```diff
@@ -0,0 +1,47 @@
+(function(exports){
+
+  var newsLog
+  var summaryrequest
+  var newsview
+
+  function newsLog(){
+    return newslog
+  }
+
+  function summaryRequest(){
+    return summaryrequest
+  }
+
+  function newsView(){
+    return newsview
+  }
+
+  function showSelectedSummary(){
+    window.addEventListener("hashchange", findSummaryUrl)
+  }
+
+  function findSummaryUrl(){
+    requestSummary(findIndex(window.location))
+  }
+
+  function findIndex(location){
+    return location.hash.split("#")[1];
+  }
+
+  function requestSummary(index){
+    var articleUrl = newslog.articles[index].url
+    summaryrequest = new ArticleSummaryRequest(articleUrl)
+    summaryrequest.requestSummary(newsView().displayArticleSummary)
+  }
+
+  function onLoad(){
+    newslog = new NewsLog()
+    newsview = new NewsHeadlinesView(newsLog())
```

Comment:
I think there is some confusion over the use of brackets - when you add () it executes the object as if it were a function. But newslog here looks like an object not a function. 

You mention in your self-review that you are not sure about callbacks yet as well - this is something lots of people struggle with. I've yet to find a really good resource that demystifies it. Maybe took a look at https://gist.github.com/amysimmons/3d228a9a57e30ec13ab1 (and let me know if that helps)